core/blockchain/block_validator/impl/block_validator_impl.cpp

```diff
@@ -150,9 +149,8 @@ namespace fc::blockchain::block_validator {
 
   outcome::result<std::reference_wrapper<BlockValidatorImpl::TipsetCPtr>>
   BlockValidatorImpl::getParentTipset(const BlockHeader &block) const {
-    IPLDBlock ipld_block = IPLDBlock::create(block);
-    if (parent_tipset_cache_
-        && parent_tipset_cache_.value().first == ipld_block.cid) {
+    auto cid{primitives::cid::getCidOfCbor(block).value()};
```

Comment:
Maybe check to error?

Reply:
Or add assert that there is no error.

Reply:
`.value()` throws `std::system_error` (like OUTCOME_EXCEPT or cbor streams)